src/Native/Unix/System.Native/pal_termios.cpp

```diff
@@ -0,0 +1,356 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+#include "pal_types.h"
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <stdio.h>
+
+/* This is dup of System/IO/Ports/NativeMethods.cs */
+enum {
+    NoneParity = 0,
+    OddParity = 1,
+    EvenParity = 2,
+    MarkParity = 3,
+    SpaceParity = 4
+};
+
+/* System/IO/Ports/Handshake.cs */
+enum {
+    HandshakeNone,          // No Flow Control
+    HandshakeSoft,          // Software Flow Control -> XOn/Xoff
+    HandshakeHard,          // Hardware Flow Control -> RTS/CTS
+    HandshakeBoth           // Software & Hardware Flow Control
+};
+
+static int  SystemNative_Termios_GetStatus(int fd)
+{
+    int status = 0;
+    if(ioctl(fd, TIOCMGET, &status) < 0)
+    {
+        return -1;
+    }
+    return status;
+}
+
+static int  SystemNative_Termios_SetStatus(int fd, int status)
+{
+    if (ioctl(fd, TIOCMSET, &status) < 0)
+    {
+        return -1;
+    }
+    return 0;
+}
+
+extern "C" int32_t SystemNative_Termios_GetDcd(int fd)
+{
+    int serial = SystemNative_Termios_GetStatus(fd);
+    if (serial >= 0)
+    {
+        return (serial & TIOCM_CAR) ? 1 : 0;
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_GetCts(int fd)
+{
+    int serial = SystemNative_Termios_GetStatus(fd);
+    if (serial >= 0)
+    {
+        return (serial & TIOCM_CTS) ? 1 : 0;
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_GetRts(int fd)
+{
+    int serial = SystemNative_Termios_GetStatus(fd);
+    if (serial >= 0)
+    {
+        return (serial & TIOCM_RTS) ? 1 : 0;
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_GetDsr(int fd)
+{
+    int serial = SystemNative_Termios_GetStatus(fd);
+    if (serial >= 0)
+    {
+        return (serial & TIOCM_DSR) ? 1 : 0;
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_GetDtr(int fd)
+{
+    int serial = SystemNative_Termios_GetStatus(fd);
+    if (serial >= 0)
+    {
+        return (serial & TIOCM_DTR) ? 1 : 0;
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_SetDtr(int fd, bool set)
+{
+    int status = SystemNative_Termios_GetStatus(fd);
+    if (status >= 0)
+    {
+        if (set)
+        {
+            status |= TIOCM_DTR;
+        }
+        else
+        {
+            status &= ~TIOCM_DTR;
+        }
+        return SystemNative_Termios_SetStatus(fd, status);
+    }
+    return -1;
+}
+
+extern "C" int32_t SystemNative_Termios_SetRts(int fd, bool set)
+{
+    int status = SystemNative_Termios_GetStatus(fd);
+    if (status >= 0)
+    {
+        if (set)
+        {
+            status |= TIOCM_RTS;
+        }
+        else
+        {
+            status &= ~TIOCM_RTS;
+        }
+        return SystemNative_Termios_SetStatus(fd, status);
+    }
+    return -1;
+}
+
+static speed_t
+SystemNative_speed2termios(int speed)
```

Comment:
Could you please keep the naming consistent - to start each part of the name with uppercase letter?